### PR TITLE
feat: create InsertEvent task for Google Calendar

### PR DESCRIPTION
**Pull request by shrutimantri**

### CODE REVIEW

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        title = "End time of the event."
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    protected CalendarTime endTime;
+
+    @Schema(
+        title = "Creator of the event."
+    )
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:70]:
```suggestion
    @PluginProperty
```

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    protected Attendee creator;
+
+    @Schema(
+        title = "List of attendees in the event."
+    )
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:76]:
```suggestion
    @PluginProperty
```

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    protected CalendarTime startTime;
+
+    @Schema(
+        title = "End time of the event."
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:64]:
```suggestion
    @PluginProperty
```

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        title = "Geographic location of the event as free-form text."
+    )
+    @PluginProperty(dynamic = true)
+    protected String location;
+
+    @Schema(
+        title = "Start time of the event."
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:57]:
```suggestion
    @PluginProperty
```

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        eventMetadata.setSummary(runContext.render(this.summary));
+        if (this.description != null) {
+            eventMetadata.setDescription(runContext.render(this.description));
+        }
+
+        if (this.location != null) {
+            eventMetadata.setLocation(runContext.render(this.location));
+        }
+
+        EventDateTime eventStartTime = new EventDateTime().setDateTime(new DateTime(startTime.dateTime)).setTimeZone(startTime.timeZone);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:131]: You forget to render it

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        eventMetadata.setStart(eventStartTime);
+
+        EventDateTime eventEndTime = new EventDateTime().setDateTime(new DateTime(endTime.dateTime)).setTimeZone(endTime.timeZone);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:134]: You forget to render it

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        eventMetadata.setEnd(eventEndTime);
+
+        if (attendees != null && attendees.size() > 0) {
+            List<EventAttendee> eventAttendees = new ArrayList<>();
+            for (Attendee attendee: attendees){
+                EventAttendee eventAttendee = new EventAttendee().setDisplayName(attendee.displayName).setEmail(attendee.email);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:140]: You forget to render it

File: src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+                eventAttendees.add(eventAttendee);
+            }
+            eventMetadata.setAttendees(eventAttendees);
+        }
+
+        if (creator != null) {
+            Creator eventCreator = new Event.Creator().setDisplayName(creator.displayName).setEmail(creator.email);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/googleworkspace/calendar/AbstractInsertEvent.java:147]: You forget to render it

**Review by loicmathieu — CHANGES_REQUESTED** [review]
Two remarks:
- AbstractInsertEvent is not necessary, you could merge it with InsertEvent (not mandatory)
- We have a new plugin property framework that is more capable that the current mechanism, you can see for ex how it works in this PR: https://github.com/kestra-io/plugin-redis/pull/73. Please use it for your future PR.

**Comment by shrutimantri** [comment]
Addressed the comments

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**